[DOCS-4939] Add a quickstart for DD agent->OPW configuration. (#17042)
DataDog/documentation: content/en/getting_started/observability_pipelines/_index.md

---
title: Getting Started with Observability Pipelines
kind: Documentation
aliases:
    - /observability_pipelines/quickstart/
further_reading:
  - link: "/observability_pipelines/production_deployment_overview/"
    tag: "Documentation"
    text: "Production deployment overview for the Observability Pipelines Worker"
  - link: "https://dtdg.co/d22op"
    tag: "Learning Center"
    text: "Safe and Secure Local Processing with Observability Pipelines"
---

## Overview

The [Observability Pipelines Worker][1] can collect, process, and route logs and metrics from any source to any destination. Using Datadog, you can build and manage all of your Observability Pipelines Worker deployments at scale. 

This guide walks you through deploying the Worker in your common tools cluster and configuring the Datadog Agent to send logs and metrics to the Worker.

> {{< img src="/observability_pipelines/quickstart/dd-pipeline.png" alt="A diagram of a couple of workload clusters sending their data through the Observability Pipelines aggregator." >}}

Reviewer: Image will be updated with Brand Design's once it's ready.

> {{< img src="getting_started/observability_pipelines/dd-pipeline.png" alt="A diagram of a couple of workload clusters sending their data through the Observability Pipelines aggregator." >}}

Reviewer: Image will be updated with Brand Design's once it's ready.

## Assumptions
* You are already using Datadog and want to use Observability Pipelines.
* Your services are deployed to a Kubernetes cluster in Amazon Elastic Kubernetes Service (EKS), Azure Kubernetes Service (AKS), or Google Kubernetes Engine (GKE).

> * You have administrative access to the cluster(s) where the Observability Pipelines Worker is going to be deployed, as well as to the workloads that are going to be aggregated.

Reviewer 2: ```suggestion
* You have administrative access to the clusters where the Observability Pipelines Worker is going to be deployed, as well as to the workloads that are going to be aggregated.
```

* You have a common tools or security cluster for your environment to which all other clusters are connected.

> # Prerequisites

Reviewer 2: ```suggestion
## Prerequisites
```

Reviewer 2: I think we should change this to a h2. If you look at the preview site, you'll see that this heading doesn't appear in the on-page TOC.

Reviewer: Oh yes, good catch!

Before installing, make sure you have:

* A valid [Datadog API key][2].
* An Observability Pipelines Configuration ID.

You can generate both of these in [Observability Pipelines][3].

To run the Worker on your Kubernetes nodes, you need a minimum of two nodes with one CPU and 512MB RAM available. Datadog recommends creating a separate node pool for the Workers, which is also the recommended configuration for production deployments.

> ## Provider-specific requirements

Reviewer 2: ```suggestion
### Provider-specific requirements
```

{{< tabs >}}
{{% tab "AWS EKS" %}}

> * The [AWS Load Balancer controller][1] is required. If you created your Amazon EKS cluster through the UI, then it is probably already installed.

Reviewer 2: Are there any additional instructions we can add here so that customers know how they can confirm whether the AWS Load Balancer controller is installed?

Reviewer: Updating with a command line that users can run to check if it is installed.

* Datadog recommends using Amazon EKS >= 1.16.

[1]: https://docs.aws.amazon.com/eks/latest/userguide/aws-load-balancer-controller.html
{{% /tab %}}
{{% tab "Azure AKS" %}}
There are no specific requirements for Azure AKS.
{{% /tab %}}
{{% tab "Google GKE" %}}
There are no specific requirements for Google GKE.
{{% /tab %}}
{{< /tabs >}}

## Installing the Observability Pipelines Worker
### Download the Helm chart

{{< tabs >}}
{{% tab "AWS EKS" %}}
Download the [Helm chart][1] for AWS EKS. 

In the Helm chart, replace the `datadog.apiKey` and `datadog.configKey` values to match your pipeline. Then, install it in your cluster with the following commands:

```

> $ helm repo add datadog https://helm.datadoghq.com

Reviewer 2: @maycmlee An idea: What if we drop the $ and use the new terminal command styling instead?

Reviewer: Good idea! Updated with new shell code block.

In the Helm chart, replace the `datadog.apiKey` and `datadog.configKey` values to match your pipeline. Then, install it in your cluster with the following commands:

```
$ helm repo add datadog https://helm.datadoghq.com
$ helm repo update
$ helm update --install \
    opw datadog/observability-pipelines-worker \
    -f google_gke.yaml
```

[1]: /resources/yaml/observability_pipelines/quickstart/google_gke.yaml
{{% /tab %}}
{{< /tabs >}}

### Load balancing
Use the load balancers provided by your cloud provider.
They adjust based on autoscaling events that the default Helm setup is configured for. The load balancers are internal-facing,
so they are only accessible inside your network.

Use the load balancer URL given to you by Helm when you configure the Datadog Agent.

#### Cross-availability-zone load balancing
The provided Helm configuration tries to simplify load balancing, but you must take into consideration the potential price implications of cross-AZ traffic. Wherever possible, the samples try to avoid creating situations where multiple cross-AZ hops can happen.

{{< tabs >}}
{{% tab "AWS EKS" %}}
NLBs provisioned by the [AWS Load Balancer Controller][1] are used.

> The sample configurations do not enable the "cross-zone load balancing" feature available in this controller. To enable it, add the following annotation to the `service` block:

Reviewer 2: ```suggestion
The sample configurations do not enable the cross-zone load balancing feature available in this controller. To enable it, add the following annotation to the `service` block:
```

The following are two important components in the example configuration:
- `logs_parse_ddtags`: Parses the tags that are stored in a string into structured data. 
- `logs_finish_ddtags`: Re-encodes the tags so that it is in the format as how the Datadog Agent would send it.

Internally, the Datadog Agent represents log tags as a CSV in a single string. To effectively manipulate these tags, they must be parsed, modified, and then re-encoded before they are sent to the ingest endpoint. These steps are written to automatically perform those actions for you. Any modifications you make to the pipeline, especially for manipulating tags, should be in between these two steps.

### Processing metrics
The provided metrics pipeline does not require additional parsing and re-encoding steps. Similar to the logs pipeline, it tags incoming metrics for traffic accounting purposes. Due to the additional cardinality, this may have cost implications for custom metrics.

> At this point, your environment is configured for Observability Pipelines with data flowing through it. Further configuration is likely required for your specific use case(s), but the tools provided gives you a starting point.

Reviewer 2: ```suggestion
At this point, your environment is configured for Observability Pipelines with data flowing through it. Further configuration is likely required for your specific use cases, but the tools provided gives you a starting point.
```
